app/src/data/host-context.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import { QuestionType } from '../models/QuestionType'
+import { ResponseValue } from '../models/ResponseValue'
+import { ApiWrapper } from './ApiWrapper'
+
+const HostDashboardContext = React.createContext({})
+const hostsFetcher = new ApiWrapper()
+
+interface HostDashboardData {
+    hostQuestions: Array<QuestionType>
+    hostResponses: Array<ResponseValue>
+    loaderState: {
+        loading: boolean
+        message: string
+    }
+}
+
+enum HostDashboardActionType {
+    BeginFetchQuestions,
+    FinishFetchQuestions,
+    GetHostById,
+    isLoading,
+    BeginPostAnswer,
+    FinishPostAnswer,
+    Error,
+}
+
+interface HostDashboardAction {
+    type: HostDashboardActionType
+    payload?: HostDashboardData | Array<QuestionType> | boolean | string
+}
+
+const initialState: HostDashboardData = {
+    hostQuestions: [],
+    hostResponses: [],
+    loaderState: {
+        loading: false,
+        message: '',
+    },
+}
+
+function hostDashboardReducer(
+    state: HostDashboardData,
+    action: HostDashboardAction
+): HostDashboardData {
+    switch (action.type) {
+        case HostDashboardActionType.BeginFetchQuestions: {
+            return {
+                ...state,
+                loaderState: {
+                    loading: true,
+                    message: action.payload as string,
+                },
+            }
+        }
+        case HostDashboardActionType.FinishFetchQuestions: {
+            return {
+                ...state,
+                loaderState: {
```

Comment:
For this case, the text "Finished Loading" will never display, since this modal will not be visible as soon as this state change is propagated to the UI. One thing we could add here (during dev/debug ONLY) would be some descriptive message to indicate that this loader's was not properly assigned. Alternatively, eliminating line 61 is acceptable.

Reply:
oops i did not realize this was for actually displaying the text! good point.